src/storage/log/mod.rs

```diff
@@ -17,7 +17,7 @@ use hashbrown::HashMap;
 /// The `BLOCK_SIZE` constant represents the size of a block used for buffering disk writes in the
 /// write-ahead log. It determines the maximum amount of data that can be held in memory before
 /// flushing to disk. Larger block sizes can improve write performance but might use more memory.
-pub(crate) const BLOCK_SIZE: usize = 32768;
+pub(crate) const BLOCK_SIZE: usize = 4 * 1024;
```

Comment:
Was there any specific reason we wanted to decrease the max aol segment size(correct?) from 32KB down to 4KB here?

I think this is only for `Durability::Eventual` and does not affect `Immediate`, and you usually want Eventual when you need more throughput in exchange of durability, which justifies larger segment size?

Reply:
So the reason I did not want to keep 32kb is because as per my observation, a sdb transaction was occupying around ~1000 bytes. I can't tell this with certainty for sure because it depends on the type of key/values being used. But the issue was that there could be a loss of too many transactions in case of an abrupt restart if the size is small. Hence I decided to keep it small, and probably make this configurable with time to improve performance.

If Durability is eventual, the transactions are just buffered as per the BLOCK_SIZE as described [here](https://github.com/surrealdb/surrealkv/blob/aca3678480fcd4329b009da584cee43da3a21a03/src/storage/log/mod.rs#L1022) and then flushed to disk. The segment size is to avoid keeping a single file for commit log, this way it is easier to do chunk upload to s3 and also for recovery, so that was the motive behind writing in chunks. I've increased this to 512mb in subsequent PRs.

Please let me know if that answers your question

Reply:
Reverted back to 32kb